### ISSUE TITLE
Some itch.io games aren't shown on the library

### ISSUE THREAD

**Issue by rdlmda** [lutris/lutris#5515]
### Bug description

I have 119 games on my itch.io library, only 90 show up on Lutris.

Some of those are non-executable media (i.e. PDF files), but some are zip files containing an executable game. For instance: https://berji.itch.io/disco-noire

Apparently Lutris uses the "traits" JSON property to populate the list, as the games are present on file but don't show up on the library

```
$ cd ~/.cache/lutris/itchio/api

$ jq '.[].game.title' profile/owned-keys.json | wc -l
$ jq 'map(select(.game.traits | index("p_windows") != null or index("p_linux") != null))' profile/owned-keys.json | jq '.[].game.title'
$ jq 'map(select(.game.traits | index("p_windows") == null and index("p_linux") == null))' profile/owned-keys.json | jq '.[].game.title'

$ cat games/*.json | jq '.title' | wc -l
$ cat games/*.json | jq 'select(.traits | index("p_windows") != null or index("p_linux") != null)' | jq '.title'
$ cat games/*.json | jq 'select(.traits | index("p_windows") == null and index("p_linux") == null)' | jq '.title'
```

### How to Reproduce

1. Visit https://itch.io/my-purchases and manually count your games
2. Sync your itch.io account succesfully
3. Check that some games are missing

### Expected behavior

All Windows and Linux games from itch.io should show up on the library

### Log output

```shell
2024-06-10 05:09:33,923: Command 'wine' not found on your system
2024-06-10 05:09:33,923: Command 'fluidsynth' not found on your system
INFO     2024-06-10 05:09:34,408 [application.do_command_line:479]:Starting Lutris 0.5.17
INFO     2024-06-10 05:09:34,547 [startup.run_all_checks:120]:NVIDIA GeForce 930MX (10de:134e 1043:1490 nvidia) Driver 535.171.04
INFO     2024-06-10 05:09:34,615 [startup.run_all_checks:120]:Intel HD Graphics 620 (8086:5916 1043:1490 i915) Driver 23.2.1
DEBUG    2024-06-10 05:09:35,120 [xrandr.get_resolutions:91]:Retrieving resolution list
DEBUG    2024-06-10 05:09:35,406 [xrandr._get_vidmodes:18]:Retrieving 48 video modes from XrandR
DEBUG    2024-06-10 05:09:35,509 [path_cache._update_missing_games:114]:Checking for missing games
DEBUG    2024-06-10 05:09:35,604 [api.check_stale_runtime_versions:52]:Modified at Mon 10 Jun 2024 04:16:21 AM , will update after Mon 10 Jun 2024 10:16:21 AM 
DEBUG    2024-06-10 05:09:35,618 [lutriswindow.create_runtime_updater_cb:1244]:Runtime up to date
DEBUG    2024-06-10 05:09:47,214 [http._request:88]:GET https://api.itch.io/profile
DEBUG    2024-06-10 05:09:48,047 [base.wipe_game_cache:441]:Deleting itchio cache /home/ruda/.cache/lutris/itchio/api/
DEBUG    2024-06-10 05:09:48,050 [base.wipe_game_cache:175]:Deleting games from service-games for itchio
DEBUG    2024-06-10 05:09:48,166 [http._request:88]:GET https://api.itch.io/profile
DEBUG    2024-06-10 05:09:48,953 [http._request:88]:GET https://api.itch.io/profile/owned-keys?page=1
DEBUG    2024-06-10 05:09:50,009 [http._request:88]:GET https://api.itch.io/profile/owned-keys?page=2
DEBUG    2024-06-10 05:09:51,100 [http._request:88]:GET https://api.itch.io/profile/owned-keys?page=3
```

### System Information

```shell
Vulkan support: YES
Esync support: YES
Fsync support: YES
Wine installed: NO
Gamescope: NO
Mangohud: YES
Gamemode: YES
Steam: YES
In Flatpak: NO
[System]
OS: Linux Mint 21.3 Virginia
Arch: x86_64
Kernel: 5.15.0-112-generic
Desktop: X-Cinnamon
Display Server: x11
[CPU]
Vendor: GenuineIntel
Model: Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
Physical cores: 2
Logical cores: 4
[Memory]
RAM: 19.4 GB
Swap: 16.0 GB
[Graphics]
Vendor: Intel
OpenGL Renderer: Mesa Intel(R) HD Graphics 620 (KBL GT2)
OpenGL Version: 4.6 (Compatibility Profile) Mesa 23.2.1-1ubuntu3.1~22.04.2
OpenGL Core: 4.6 (Core Profile) Mesa 23.2.1-1ubuntu3.1~22.04.2
OpenGL ES: OpenGL ES 3.2 Mesa 23.2.1-1ubuntu3.1~22.04.2
Vulkan Version: 1.3.204
Vulkan Drivers: NVIDIA GeForce 930MX (1.3.242), Intel(R) HD Graphics 620 (KBL GT2) (1.3.255)
```

**Comment by danieljohnson2** [comment]
That's right, Lutris filters the games based on the traits; looks like it wants a Linux or Windows game. This is how it knows how to install the game.

But is not Disco Noire a Windows game? Or what it is?

Can you provide the .json file for this game?

**Comment by rdlmda** [comment]
Yes, Disco Noire is a Windows game. Here's its JSON file:

```
{
  "short_text": "A point-and-click detective noire game.",
  "type": "default",
  "created_at": "2022-01-10T20:28:28.000000000Z",
  "min_price": 100,
  "title": "Disco Noire",
  "url": "https://berji.itch.io/disco-noire",
  "published_at": "2022-02-19T05:57:26.000000000Z",
  "traits": [
    "can_be_bought"
  ],
  "user": {
    "url": "https://berji.itch.io",
    "id": 1648232,
    "username": "Berji"
  },
  "id": 1348837,
  "cover_url": "https://img.itch.zone/aW1nLzc4NTEwMDEucG5n/315x250%23c/2UQL7%2F.png",
  "classification": "game",
  "download_key_id": 71666506
}
```

**Comment by rdlmda** [comment]
I just noticed the itch.io interface doesn't show the platform icon for Disco Noire:

![image](https://github.com/lutris/lutris/assets/4602525/9c7cbcc0-d123-471e-a6c4-1d615c950631)

In contrast with other games, notice the icons:

![image](https://github.com/lutris/lutris/assets/4602525/633cf16d-813f-4f66-abf7-4f9332455cdb)

I'm assuming Lutris is just parsing auhtor (or vendor) provided information.

Is there any way to "manually" import a game into the itch library, without having to download the zip file, uncompress it, install and then add it as a standalone game?

**Comment by danieljohnson2** [comment]
Yes, the 'trait' indicating a Windows game is missing, and it seems like it ought to be there.

There's no way to import that game in Lutris, but I want to know what would happen so [please try this branch I just made up](https://github.com/lutris/lutris/tree/dj/traitless).

No need to install; just download it and run the ``bin/lutris`` script, and refresh the Itch.io source. It should now think that every non-Linux game is a Windows game.

Does this allow you to install Disco Noire? Does it think your PDFs are games now, and if so, what happens if you install one? Can you play it? Is it fun?

**Comment by rdlmda** [comment]
Ufortunately I can't isntall or run any Windows game (or even a Windows / Wine based lutris component as the EA app) because #5507

**Comment by danieljohnson2** [comment]
Please give it a try. The EA app thing is the EA app installer failing, and Itch.io doesn't have anything like that. It might work.

**Comment by Glog78** [comment]
https://tobyfox.itch.io/deltarune << this seems to be another very popular and free game.
I have it in my Lutris list but it's not shown up and therefor not installable.

**Comment by danieljohnson2** [comment]
I can check that one. DeltaRune does not list any platforms in its 'traits' at all; Lutris has no idea how to install this. Is it a Windows game? A Linux game? No-one knows for sure!

**Comment by Glog78** [comment]
Wouldn't be the solution to use the default downloader / installer ... i mean in  alot of games i have on lutris there is only the wine default installer and i need to set by hand the executable and the path anyway ...

**Comment by danieljohnson2** [comment]
Lutris could treat a game with no platforms have have a default set of 'Linux, Windows' and give you the choice of which to install.

But I think this might not be wise; there's a bunch of non-game stuff on itch.io, and treating them all as games is going to be very confusing.

**Comment by Glog78** [comment]
I can't make this decision for you guys. I can only make some suggestions which would work for me. And yes there is alot of download only stuff on lutris too and you also need consider the mac people ...
There is also a more "complicated" workaround. You use the itch.io app to download the stuff and add it to a runner on lutris with its own problems ... (no updates / two clients) 
For me it would be cool if i could download (even non gaming content -> ebooks / pdfs / video) using lutris and if needed adjust the runner after.

**Comment by mainrs** [comment]
> But I think this might not be wise; there's a bunch of non-game stuff on itch.io, and treating them all as games is going to be very confusing.

Items on itch.io are categorized by their type. Only video games should have the a classification of `game`. This is the API response for the game OP mentioned:

```json
{
    "game": {
        "classification": "game",
        "published_at": "2022-02-19T05:57:26.000000000Z",
        "min_price": 299,
        "sale": {
            "rate": 69,
            "end_date": "2025-06-01T16:00:00.000000000Z",
            "id": 153896,
            "start_date": "2025-05-29T16:42:00.000000000Z"
        },
        "title": "Disco Noire",
        "user": {
            "url": "https://berji.itch.io",
            "username": "Berji",
            "id": 1648232
        },
        "short_text": "A point-and-click detective noire game.",
        "cover_url": "https://img.itch.zone/aW1nLzc4NTEwMDEucG5n/315x250%23c/2UQL7%2F.png",
        "traits": [
            "can_be_bought"
        ],
        "created_at": "2022-01-10T20:28:28.000000000Z",
        "type": "default",
        "id": 1348837,
        "url": "https://berji.itch.io/disco-noire"
    }
}
```

Lutris could assume that an item can be installed if it is classified as a game. It would have to decide if it is windows or linux at runtime, since there is no way to know that before unzipping the file

**Comment by danieljohnson2** [comment]
OK, I guess that pushes me over the edge. I've committed a fix that now:

1. Includes only games (classification=game or no classification)
2. If there are no ``p_`` traits at all, assume the Linux and Wine runners will work
3. ... but still list no platforms for the game.
4. Logs a warning for each game that is handled this way.

This means a game that has ``p_osx`` only will still be filtered out, as Lutris does not have support for running OS X games.

That should get _Disco Noire_ to appear.

@rdlmda, can you test this? Download [the master branch](https://github.com/lutris/lutris) and run the ``bin/lutris`` script within - no need to install. Just reload your itch.io library; do you get more of your games? All of them?

**Comment by mainrs** [comment]
@danieljohnson2 It works! The game (and others) are now part of my library. I bought the same and checked if the launch configuration is correct. Sadly, lutris selects the wrong exe.

```console
disco-noire/drive_c/disco-noire 
❯ ls -gahl
Permissions Size User Group Date Modified Name
.rw-r--r--  654k me   me    10 Jan  2022   'Disco Noire - Moonjam 2022.exe'
drwx------     - me   me    10 Jan  2022   'Disco Noire - Moonjam 2022_Data'
drwx------     - me   me    10 Jan  2022   MonoBleedingEdge
.rw-r--r--  1.1M me   me    15 Jun  2021   UnityCrashHandler64.exe
.rw-r--r--   28M me   me    15 Jun  2021   UnityPlayer.dll
```

Lutris tries to launch `UnityCrashHandler64.exe`. Maybe trying to find an exe that partially matches the title would work? And if none is found, filter out the list for known executable names that are part of engines but not the binary itself (like UnityCrashHandler64.exe). And if there is still more than one left, just select the first one alphabetically?

Thanks for taking your time and working on this! <3

**Comment by danieljohnson2** [comment]
Yeah, it's not a a perfect world. Lutris does not know which .exe is the right one; it prefers a link if it finds, one, then a 64-bit .exe, then a 32-bit .exe; it doesn't care about titles.

Can you fix this by manually updating the game executable?

**Comment by mainrs** [comment]
I manually changed it earlier and it works.
